### PR TITLE
[FIX] UI: Improve documentation about plugin manipulations

### PR DESCRIPTION
**Pull request by thibsy**
Hi folks,

This PR improves the documentation about manipulations plugins can make to the UI framework. Primarily, I ...

- added some insight about the **rendering chain** plugin developers need to worry about, because I have stumbled over this several times myself now :).
- added some documentation about introducing custom UI components, because I think this is a powerful functionality which needs some more recognition, along with a code example.
- improved the existing example because it featured some wrong comments and lacked detail.
- improved the document grammatically.
- removed the contribution section, because we have VCS for this.

P.S. Unsure about the proper ILIAS copyright license headers in examples.

Kind regards,
@thibsy

### CODE REVIEW

File: docs/development/ui-plugin-manipulations.md [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
-- remove entire components from the ILIAs structure
+- `ilPlugin::exchangeUIRendererAfterInitialization()`
+- `ilPlugin::exchangeUIFactoryAfterInitialization()`
+
+While these functions offer a lot of opportunities, they also come with a lot of responsibility, since it allows the
+developers to perform major interventions in the build and rendering process. For example a developer can...
+
+- prohibit certain plugins from doing changes
+- remove entire components from the ILIAS structure
+- introduce entirely components
```
Comment (klees) [inline docs/development/ui-plugin-manipulations.md]:
```suggestion
- introduce completely new components
```

File: docs/development/ui-plugin-manipulations.md [hunk 2]
```diff
@@ -30,27 +35,31 @@ The first step on the way to make a UI change is to ask the following question:
 
 **Is the desired change structural or visual?**
 
-Structural changes do affect the UI by restructuring component without affecting the inner HTMl od the components themselves.
-This kind of changes should be made inside the **exchangeUIFactoryAfterInitialization** function.
-Proper examples are:
+Structural changes affect the UI by restructuring component without affecting the inner HTMl of the components
```
Comment (klees) [inline docs/development/ui-plugin-manipulations.md]:
```suggestion
Structural changes affect the UI by restructuring component without affecting the inner HTML of the components
```

**Review by klees — APPROVED** [review]
I really like it. Thanks a lot!

**Comment by Amstutz** [comment]
thx a lot! cherry picking seems to create issues. ill look into it.

**Pull request merged**